test/burn/org/apache/cassandra/concurrent/LongSharedExecutorPoolTest.java

```diff
@@ -135,19 +139,27 @@ private void testPromptnessOfExecution(long intervalNanos, float loadIncrement)
         {
             if (System.nanoTime() > until)
             {
-                System.out.println(String.format("Completed %.0fK batches with %.1fM events", runs * 0.001f, events * 0.000001f));
+                logger.info(String.format("Completed %.0fK batches with %.1fM events", runs * 0.001f, events * 0.000001f));
                 events = 0;
                 until = System.nanoTime() + intervalNanos;
                 multiplier += loadIncrement;
-                System.out.println(String.format("Running for %ds with load multiplier %.1f", TimeUnit.NANOSECONDS.toSeconds(intervalNanos), multiplier));
+                logger.info(String.format("Running for %ds with load multiplier %.1f", TimeUnit.NANOSECONDS.toSeconds(intervalNanos), multiplier));
             }
 
             // wait a random amount of time so we submit new tasks in various stages of
             long timeout;
+            boolean timeoutIsMax = false;
             if (pending.isEmpty()) timeout = 0;
-            else if (Math.random() > 0.98) timeout = Long.MAX_VALUE;
+            else if (Math.random() > 0.98)
+            {
+                timeout = System.nanoTime() + TimeUnit.HOURS.toNanos(1);
+                timeoutIsMax = true;
+            }
             else if (pending.size() == executorCount) timeout = pending.first().timeout;
-            else timeout = (long) (Math.random() * pending.last().timeout);
+            else timeout = System.nanoTime() + (long) (Math.random() * (pending.last().timeout - System.nanoTime()));
+
+            if ((timeout - System.nanoTime()) < 0)
```

Comment:
Should we not handle the casew when `pending.isEmpty()` in some special way?
What does it actually mean from test perspective?

In such case timeout will be set to 0 and warning will be issued, which is bit strange as it is perfectly expected.

Reply:
It seems it is a bit convoluted because for the first iteration the first part is skipped, so I guess we should switch the first part and the second part of the loop.